serviceerror/convert.go

```diff
@@ -42,6 +42,14 @@ func ToStatus(err error) *status.Status {
 	if svcerr, ok := err.(ServiceError); ok {
 		return svcerr.Status()
 	}
+	// err does not implement ServiceError directly, but check if it wraps it.
+	// This path does more allocation so prefer to return a ServiceError directly if possible.
+	var svcerr ServiceError
+	if errors.As(err, &svcerr) {
```

Comment:
I support this change because this is what https://pkg.go.dev/google.golang.org/grpc/status does. But we should have never had a `ServiceError` IMO because `GRPCStatus()` should have been the method and existing gRPC status package should have been good enough. But I guess that ship sailed long ago.

Technically it's a backwards incompatible change, but from an SDK POV there are no obvious user-facing times where this will accidentally return a status where it wouldn't have otherwise because user-facing deserialized error hierarchy only doesn't contain this type at any depth (it's not a failure type).